xbmc/pvr/recordings/PVRRecording.cpp

```diff
@@ -489,9 +497,25 @@ bool CPVRRecording::IsInProgress() const
   //       Only the state of the related timer is a safe indicator that the backend
   //       actually is recording this.
 
-  return CServiceBroker::GetPVRManager().Timers()->HasRecordingTimerForRecording(*this);
+  if (m_iChannelUid != PVR_CHANNEL_INVALID_UID && m_iRecordingState != PVR_RECORDED_TV_ITEM_STATE_INVALID)
+  {
+    return (m_iRecordingState == PVR_RECORDED_TV_ITEM_STATE_IN_PROGRESS);
+  }
+  else
+  {
+    return CServiceBroker::GetPVRManager().Timers()->HasRecordingTimerForRecording(*this);
+  }
+}
+
+/* Or with 'ChannelUid' and 'RecordingState' mandatory it becomes:
```

Comment:
Not related to channel uid being mandatory. But as we need to make recordingstate mandatory, you can delete the ols impl. and straight activate the new, simple impl.

Reply:
> Not related to channel uid being mandatory. But as we need to make recordingstate mandatory, you can delete the old impl. and straight activate the new, simple impl.

Without ChannelUid and only RecordingState implemented then recordings in progress show red dot but 'stop recording' cannot resolve correct timer and so fails.

Reply:
Don't hack around until you see on the screen what you expect. ;-)
If recordingstate is mandatory, there is no need for Recording::IsInProgress relying on channel uid being available.
If this causes problems, those must be solved differently.